app/views/portal/show.rb

```diff
@@ -455,152 +455,98 @@ def merge_values(fields, separator = ' ')
-                if !extra_val
-                  next
-                end
-                if xtra[:format_date]
-                  begin
-                    date = Time.parse(extra_val)
-                    formatted = date.strftime(xtra[:format_date])
-                    extra_val = formatted
-                  rescue
-                  end
-                end
-                extra_info_builder = extra_info
-                path_segments = xtra[:map_to] || xtra[:field]
-                path_segments = path_segments.split('.')
-
-                path_segments.each.map do |path_segment|
-                  is_last = path_segment == path_segments.last
-                  extra_info_builder[path_segment] ||= (is_last ? extra_val : {})
-                  extra_info_builder = extra_info_builder[path_segment]
-                end
-                ob['extra_info'] = extra_info
+              if override[:field_url]
+                item[:url] = override[:field_url]
               end
             end
-
-            ob['text'] = text
-            subsection << ob unless text.nil? || text.blank?
           end
 
-          if subsection.size > 0
-            section_data << subsection
-            section_labels << (section[:title].nil? ? false : t(section[:title]))
+          # extra info on last
+          if val == field_values.last && !section[:extra].nil?
+            item[:extra_info] = data_section_nested_hash(section[:extra])
           end
         end
       end
 
-      {
-        title: t(data[:title]),
-        sections: section_data.each_with_index.map do |subsection, index|
-          if subsection.size > 0
-            {
-              title: section_labels[index],
-              items: subsection
-            }
-          else
-            false
-          end
-        end
-      } unless section_data.size == 0
+      subsection.reject { |item| item[:text].blank? }
     end
 
-    # def content_object_download
-    #   links = []
+    def data_section(data)
+      sections = data[:sections].map do |section|
+        {
+          title: section[:title].nil? ? false : t(section[:title]),
+          items: data_section_field_subsection(section)
+        }
+      end
 
-    #   if edm_is_shown_by_download_url.present?
-    #     links << {
-    #       text: t('site.object.actions.download'),
-    #       url: edm_is_shown_by_download_url
-    #     }
-    #   end
+      sections.reject! { |section| section[:items].blank? }
 
-    #   return nil unless links.present?
+      sections.blank? ? nil : {
+        title: t(data[:title]),
+        sections: sections
+      }
+    end
 
-    #   {
-    #     primary: links.first,
-    #     secondary: {
-    #       items: (links.size == 1) ? nil : links[1..-1]
-    #     }
-    #   }
-    # end
+    ##
+    # Creates a nested hash of field values for Mustache template
+    def data_section_nested_hash(mappings)
```

Comment:
Assignment Branch Condition size for data_section_nested_hash is too high. [16.28/15]<br>Method has too many lines. [15/10]